eng/validation/remove-oldbranches.ps1

```diff
@@ -0,0 +1,120 @@
+$testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
+New-Item -Path $testRoot -ItemType Directory | Out-Null
+
+$global:arcadeSdkPackageName = 'Microsoft.DotNet.Arcade.Sdk'
+$global:arcadeSdkVersion = $GlobalJson.'msbuild-sdks'.$global:arcadeSdkPackageName
+$global:azdoOrg = $azdoOrg
+$global:azdoProject = $azdoProject
+$global:azdoToken = $azdoToken
+$global:user = $user
+$global:barToken = $barToken
+$global:azdoRepoName = $azdoRepoName
+$global:azdoUri = "https://${global:user}:${global:azdoToken}@dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_git/${global:azdoRepoName}"
+$global:branchNamePrefix = "dev/" + $global:user + "/arcade-"
+$global:darcAzDORepoName = "https://dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_git/${global:azdoRepoName}"
+$global:lastBranch = "refs/heads/" + $global:branchNamePrefix + $global:arcadeSdkVersion
+
+function Get-AzDOHeaders()
+{
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${global:azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    return $headers
+}
+
+function Get-AzDOGitReposUri(
+    [string] $queryStringParameters
+)
+{
+    $uri = "https://dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_apis/git/repositories/${global:azdoRepoName}/refs"
+    $uri += "?api-version=5.1" + $queryStringParameters
+
+    return $uri
+}
+
+function Get-Branches()
+{
+    $queryStringParameters = "&filter=heads/&filterContains=${global:branchNamePrefix}"
+    $uri = (Get-AzDOGitReposUri -queryStringParameters $queryStringParameters)
+    $headers = Get-AzDOHeaders
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Method Get 
+    return ($content | ConvertFrom-Json).value
+}
+
+function Remove-AzDOBranches($body)
+{ 
+    $uri = Get-AzDOGitReposUri
+    $headers = Get-AzDOHeaders
+
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Body (ConvertTo-Json $body) -Method Post 
+    return ($content | ConvertFrom-Json).value
+}
+
+Write-Host "Getting remote branches for '${global:azdoRepoName}' on Azure DevOps"
+$remoteBranches = (Get-Branches)
+
+if($null -ne $remotebranches)
+{
+    $jsonBodyArray = @()
+
+    foreach($remoteBranch in $remoteBranches)
+    {
+        $branchName = $remoteBranch.name
+
+        # we want to retain the branch with the current version of Arcade in case it's still being used for investigation purposes. 
+        if($branchName -ne $global:lastBranch)
+        {
+            $oldObjectId = $remoteBranch.objectId
+            $json = @{
+                "name"=$branchName;
+                "oldObjectId"=$oldObjectId;
+                "newObjectId"="0000000000000000000000000000000000000000"
+            }
+            
+            $jsonBodyArray += $json
+
+            Write-Host "Sending '${branchName}' to AzDO API to be deleted"
```

Comment:
It looks like this log statement should be somewhere around LN 97

Reply:
I was trying to figuring out wording here to say "Identified this branch as something that needs to be deleted, so I'm configuring the body of the API request to take in this branch." :)

Reply:
ahh I see what you're doing here. Still a little confusing as we're not deleting them yet, and might cause some confusion when examining the logs if something seems off.

Reply:
I'll try to find better wording for this log.

Reply:
How about something like:

```
Gathering branches to be deleted:
Found branch <blah>. Deleting default channel
```
loop finishes
```
Deleting branches...